ddtrace/contrib/aiokafka/patch.py

```diff
@@ -0,0 +1,248 @@
+        super(TracedAIOKafkaProducerMixin, self).__init__(*args, **kwargs)
+        self._dd_bootstrap_servers = kwargs.get("bootstrap_servers")
+
+
+class TracedAIOKafkaConsumerMixin:
+    def __init__(self, *args, **kwargs):
+        super(TracedAIOKafkaConsumerMixin, self).__init__(*args, **kwargs)
+
+
+class TracedAIOKafkaProducer(TracedAIOKafkaProducerMixin, aiokafka.AIOKafkaProducer):
+    pass
+
+
+class TracedAIOKafkaConsumer(TracedAIOKafkaConsumerMixin, aiokafka.AIOKafkaConsumer):
+    pass
+
+
+def patch():
+    if getattr(aiokafka, "_datadog_patch", False):
+        return
+    aiokafka._datadog_patch = True
+
+    aiokafka.AIOKafkaProducer = TracedAIOKafkaProducer
+    aiokafka.AIOKafkaConsumer = TracedAIOKafkaConsumer
+
+    trace_utils.wrap(TracedAIOKafkaProducer, "send", traced_send)
+    trace_utils.wrap(TracedAIOKafkaProducer, "send_and_wait", traced_send_and_wait)
+    trace_utils.wrap(TracedAIOKafkaConsumer, "getone", traced_getone)
+    trace_utils.wrap(TracedAIOKafkaConsumer, "getmany", traced_getmany)
+    trace_utils.wrap(TracedAIOKafkaConsumer, "commit", traced_commit)
+
+    Pin().onto(aiokafka.AIOKafkaProducer)
+    Pin().onto(aiokafka.AIOKafkaConsumer)
+
+
+def unpatch():
+    if getattr(aiokafka, "_datadog_patch", False):
+        aiokafka._datadog_patch = False
+
+    if trace_utils.iswrapped(TracedAIOKafkaProducer.send):
+        trace_utils.unwrap(TracedAIOKafkaProducer, "send")
+    if trace_utils.iswrapped(TracedAIOKafkaProducer.send_and_wait):
+        trace_utils.unwrap(TracedAIOKafkaProducer, "send_and_wait")
+    if trace_utils.iswrapped(TracedAIOKafkaConsumer.getone):
+        trace_utils.unwrap(TracedAIOKafkaConsumer, "getone")
+    if trace_utils.iswrapped(TracedAIOKafkaConsumer.getmany):
+        trace_utils.unwrap(TracedAIOKafkaConsumer, "getmany")
+    if trace_utils.iswrapped(TracedAIOKafkaConsumer.commit):
+        trace_utils.unwrap(TracedAIOKafkaConsumer, "commit")
+
+    aiokafka.AIOKafkaProducer = _AIOKafkaProducer
+    aiokafka.AIOKafkaConsumer = _AIOKafkaConsumer
+
+
+async def traced_send(func, instance, args, kwargs):
+    pin = Pin.get_from(instance)
+    if not pin or not pin.enabled():
+        return await func(*args, **kwargs)
+
+    topic = get_argument_value(args, kwargs, 0, "topic")
+    value = get_argument_value(args, kwargs, 1, "value", True) or None
+    key = get_argument_value(args, kwargs, 2, "key", True) or ""
+    partition = get_argument_value(args, kwargs, 3, "partition", True) or None
+    headers = get_argument_value(args, kwargs, 5, "headers", True) or []
+
+    with pin.tracer.trace(
+        schematize_messaging_operation(kafkax.PRODUCE, provider="kafka", direction=SpanDirection.OUTBOUND),
+        service=trace_utils.ext_service(pin, config.kafka),
+        span_type=SpanTypes.WORKER,
+    ) as span:
+        core.dispatch("aiokafka.produce.start", (instance, topic, value, key, headers, span))
+        span.set_tag_str(MESSAGING_SYSTEM, kafkax.SERVICE)
```

Comment:
The logic in this block is similar enough to the existing logic for tagging Kafka spans that I recommend sharing existing code via a new utils function rather than copypasting.

Reply:
I would first like to make sure the logic fully works (and I also understand it) with the `core.context_with_data`. After that I'll have more knowledge of what to extract.

Reply:
Fixed